lib/cfn-nag/custom_rules/LambdaFunctionCloudWatchLogsRule.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/util/wildcard_patterns'
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class LambdaFunctionCloudWatchLogsRule < BaseRule
+  IAM_CREATEGROUP_PATTERNS = wildcard_patterns('CreateLogGroup').map! { |x| 'logs:' + x } + ['*']
+  IAM_CREATESTREAM_PATTERNS = wildcard_patterns('CreateLogStream').map! { |x| 'logs:' + x } + ['*']
+  IAM_PUTEVENT_PATTERNS = wildcard_patterns('PutLogEvent').map! { |x| 'logs:' + x } + ['*']
+
+  def rule_text
+    'Lambda functions require permission to write CloudWatch Logs'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W55'
+  end
+
+  def audit_impl(cfn_model)
+    # Iterate over each Lambda function
+    lambda_functions = cfn_model.resources_by_type('AWS::Lambda::Function')
```

Comment:
i know the SAM support is a mess, but does this logic have any relevance to the SAM function?  the SAM function might have the base roles in place already i guess?

Reply:
Right, the SAM transformation creates an execution role with correct permissions by default.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/transform-aws-serverless.html